File: week02/day03/src/Printer.java
/**
 * Created by Bálint on 2017. 03. 22..
 */
public class Printer {
  //  Create the usual class wrapper
  //  and main method on your own.

  // - Create a function called `printer`
  //   which prints the input String parameters
  //   (can have multiple number of arguments)
  public static void main(String[] args) {
    printerFunction("blabla", " yo yo", " haha");
  }

  public static void printerFunction(String... text) {
    for (String current : text) {
      System.out.print(current);
    }

  }
}

File: javascript_practice/2-functions-and-arrays/printer.js
'use strict';
// - Create a function called `printer`
//   which logs to the console the input parameters
//   (can have multiple number of arguments)

function printer() {
  for (var i = 0; i < arguments.length; i++) {
    console.log(arguments[i]);
  }
}

function printerWithForEach() {
  var args = Array.from(arguments);

  args.forEach(function(element) {
    console.log(element);
  })
}

function printerWithForEach2(...args) {
  args.forEach(function(element) {
    console.log(element);
  })
}

printer('labda', 2, 4, true);
console.log('==================');
printerWithForEach('labda', 2, 4, true);
console.log('==================');
printerWithForEach2('labda', 2, 4, true);
